server/routes/transaction.routes.js

```diff
@@ -0,0 +1,147 @@
+import {
+  addTransactionSchema,
+  updateTransactionSchema,
+  getTransactionsByPropertySchema,
+} from '../schemas/transaction.schema';
+import TransactionController from '../controllers/transaction.controllers';
+
+const router = express.Router();
+
+/**
+ * @swagger
+ * /transaction/add:
+ *   post:
+ *     tags:
+ *       - Transaction
+ *     description: Adds a new transaction
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Transaction'
+ *      description: Add a new transaction
+ *     responses:
+ *      '201':
+ *        description: Transaction added
+ *      '400':
+ *        description: Error adding transaction
+ *      '500':
+ *       description: Internal server error
+ */
+router.post(
+  '/add',
+  authenticate,
+  isAdmin,
+  schemaValidation(addTransactionSchema),
+  TransactionController.add,
+);
+
+/**
+ * @swagger
+ * /transaction/all:
+ *   get:
+ *     tags:
+ *       - Transaction
+ *     description: Get all transactions in db
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Transaction'
+ *      description: Get all transactions in db
+ *     responses:
+ *      '200':
+ *        description: returns object of transactions
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/all', authenticate, isAdmin, TransactionController.getAll);
+
+/**
+ * @swagger
+ * /transaction/update:
+ *   put:
+ *     tags:
+ *       - Transaction
+ *     description: Updates payment date of existing transaction
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Transaction'
+ *      description: Updates payment date of existing transaction
+ *     responses:
+ *      '200':
+ *        description: Transaction updated
+ *      '404':
+ *        description: Transaction not found
+ */
+router.put(
+  '/update',
+  authenticate,
+  isAdmin,
+  schemaValidation(updateTransactionSchema),
+  TransactionController.update,
+);
+
+/**
+ * @swagger
+ * /transaction/user:
+ *   get:
+ *     tags:
+ *       - Transaction
+ *     description: Get all transactions made by a user
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Transaction'
+ *      description: Get all transactions made by a user
+ *     responses:
+ *      '200':
+ *        description: returns object of transactions
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/user', authenticate, TransactionController.getAllPersonal);
+
+/**
+ * @swagger
+ * /transaction/details:
```

Comment:
This is not getting the transaction, rather it is getting all transactions for a given property. Imho, I feel this is better as a get request and the endpoint can be updated to reflect that it is for property. 

In the nearest future, we will need to create a transaction details endpoint that gets the transaction details of a single transaction based on the transaction Id.

```suggestion
 * /transaction/property/:propertyId:
```